### ISSUE TITLE
clientFactory is not a function

### ISSUE THREAD

**Issue by abovedave** [sanity-io/sanity#4007]
Upgrading form 3.0.6 to 3.1.0. I get the following error when running `npm run dev`. I believe the change relates to #3939.

<img width="1338" alt="Screenshot 2022-12-21 at 19 58 29" src="https://user-images.githubusercontent.com/649231/208992536-fa8a2edc-4b97-4f34-934c-dff9fbbee6da.png">

[Following the guidance here on setting up SSO](https://www.sanity.io/docs/auth-api-reference), I have the following config:

```typescript
import { defineConfig, createAuthStore } from 'sanity'

export default defineConfig({
  ...,
  auth: createAuthStore({
    dataset: 'production',
    projectId: '...',
    mode: 'replace',
    redirectOnSingle: true,
    providers: [
      {
        name: 'saml',
        title: '...',
        url: '...'
      }
    ]
  })
})
```

The typescript error from VS Code in 3.1.0 on `createAuthStore()` is:

```
Argument of type '{ dataset: string; projectId: string; mode: "replace"; redirectOnSingle: true; providers: { name: string; title: string; url: string; }[]; }' is not assignable to parameter of type 'AuthStoreOptions'.
  Property 'clientFactory' is missing in type '{ dataset: string; projectId: string; mode: "replace"; redirectOnSingle: true; providers: { name: string; title: string; url: string; }[]; }' but required in type 'AuthStoreOptions'.ts(2345)
index.d.ts(662, 3): 'clientFactory' is declared here.
```

**Comment by mariuslundgard** [comment]
This is indeed a bug, thanks for reporting.

You may work around it like this:
```ts
import createSanityClient from '@sanity/client'
import { defineConfig, createAuthStore } from 'sanity'

export default defineConfig({
  ...,
  auth: createAuthStore({
    clientFactory: createSanityClient, // 👈
    dataset: 'production',
    projectId: '...',
    mode: 'replace',
    redirectOnSingle: true,
    providers: [
      {
        name: 'saml',
        title: '...',
        url: '...'
      }
    ]
  })
})
```

**Comment by mariuslundgard** [comment]
Fixed in v3.1.1: https://github.com/sanity-io/sanity/releases/tag/v3.1.1
